rdrf/rdrf/export_import/definitions.py

```diff
@@ -0,0 +1,205 @@
+
+    all_types = (CDES, REFDATA, REGISTRY_DEF, REGISTRY_WITH_DATA)
+
+    registry_types = (REGISTRY_DEF, REGISTRY_WITH_DATA)
+    registry_types_names = map(lambda t: t.name, registry_types)
+    registry_types_codes = map(lambda t: t.code, registry_types)
+
+    @classmethod
+    def from_code(cls, code):
+        for t in cls.all_types:
+            if t.code == code:
+                return t
+        raise Exception("Invalid export type code '%s'" % code)
+
+    @classmethod
+    def from_name(cls, name):
+        for t in cls.all_types:
+            if t.name == name:
+                return t
+        raise Exception("Invalid export type name '%s'" % name)
+
+
+# TODO this is far from ideal and quite a bit naive
+# It is used to filter what to import if the user decides to request an import
+# type that is a subset of the export file (ex. import CDES only from a
+# Reference Data export)
+# These dependencies between data groups should be declared in this file and
+# the importer should deduce them from the definitions.
+META_FILTERS = {
+    EXPORT_TYPES.CDES: lambda m: m['name'] == 'CDEs',
+    EXPORT_TYPES.REFDATA: lambda m: m['name'] == 'Reference Data',
+    EXPORT_TYPES.REGISTRY_DEF: lambda m: m['name'] != 'Registry Data',
+}
+
+
+# Export definitons for different type of exports
+
+
+CDE_EXPORT_DEFINITION = ExportDefinition(
+    type=EXPORT_TYPES.CDES,
+    exporters_catalogue=_MAIN_CATALOGUE,
+    datagroups=(_CDE_GROUP,)
+)
+
+REFDATA_EXPORT_DEFINITION = ExportDefinition(
+    type=EXPORT_TYPES.REFDATA,
+    exporters_catalogue=_MAIN_CATALOGUE,
+    datagroups=(_REFDATA_GROUP,)
+)
+
+REGISTRY_DEF_EXPORT_DEFINITION = ExportDefinition(
+    type=EXPORT_TYPES.REGISTRY_DEF,
+    exporters_catalogue=Catalogue(
+        datagroup_exporters.catalogue,
+        model_exporters.registry_catalogue,
+        mongo_collection_exporters.catalogue),
+    datagroups=(
+        _REFDATA_GROUP,
+        _USERS_AND_USERGROUPS_GROUP,
+        _CDE_GROUP,
+        _REGISTRY_DEF_GROUP,
+    ),
+)
+
+REGISTRY_WITH_DATA_EXPORT_DEFINITION = ExportDefinition(
+    type=EXPORT_TYPES.REGISTRY_WITH_DATA,
+    exporters_catalogue=Catalogue(
+        datagroup_exporters.catalogue,
+        model_exporters.registry_catalogue,
+        mongo_collection_exporters.catalogue),
+    datagroups=(
+        _REFDATA_GROUP,
+        _USERS_AND_USERGROUPS_GROUP,
+        _CDE_GROUP,
+        _REGISTRY_DEF_GROUP,
+        GroupDefinition(
+            name='Registry Data',
+            dirname='registry_data',
+            datagroups=(
+                GroupDefinition(
+                    name='Demographic Data',
+                    dirname='demographic_data',
+                    models = (
+                        # Leave RDRFContexts before Patients, so that no
+                        # default contexts are created on import. This works
+                        # because there is no FK from RDRFContext to Patient
+                        'rdrf.RDRFContext',
```

Comment:
Files aren't exported in this version.
The JIRA issue has a TODO list, files is one item on that list:
https://ccgmurdoch.atlassian.net/browse/RDR-1215